frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/CheckUserExistsHandlerTest.java

```diff
@@ -441,4 +442,29 @@ private ClientSession getClientSession() {
         return new ClientSession(
                 authRequest.toParameters(), null, mock(VectorOfTrust.class), CLIENT_NAME);
     }
+
+    private void setupUserProfileAndClient(UserProfile userProfile) {
+        when(authenticationService.getOrGenerateSalt(userProfile)).thenReturn(SALT.array());
+        when(authenticationService.getUserProfileByEmailMaybe(EMAIL_ADDRESS))
+                .thenReturn(Optional.of(userProfile));
+        when(clientService.getClient(CLIENT_ID)).thenReturn(Optional.of(generateClientRegistry()));
+        when(clientSessionService.getClientSessionFromRequestHeaders(any()))
+                .thenReturn(Optional.of(getClientSession()));
+    }
+
+    private APIGatewayProxyResponseEvent createUserExistsApiResponse() {
```

Comment:
This is a nice tidy in general. I'd be inclined to keep this to just creating the request object (so everything except `handler.handleRequest`). And keep the `handler.handleRequest` in the tests. 

Rationale being, it's nice to structure tests as:
* Arrange
* Act
* Assert
(see e.g. [here](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80))

And to me, handler.handleRequest is the "act" bit of this, whereas in this method we're mostly involved in arranging (e.g. setting up the event). So I'd want to keep the "act" line in the tests so that it's clear what we're actually testing